Source/ModelGroups/Mesh_PrimitiveVertexColor.cs

```diff
@@ -1,76 +1,137 @@
+                // Apply the common properties to the gltf. 
+                meshPrimitive.Colors = vertexColors;
+                meshPrimitive.TextureCoordSets = null;
 
-            return combos;
-        }
+                // Apply the properties that are specific to this gltf.
+                setProperties(properties, meshPrimitive);
 
-        public Runtime.GLTF SetModelAttributes(Runtime.GLTF wrapper, Runtime.Material material, List<Property> combo, ref glTFLoader.Schema.Gltf gltf)
-        {
-            // Remove the base model's UV0
-            wrapper.Scenes[0].Nodes[0].Mesh.MeshPrimitives[0].TextureCoordSets.RemoveAt(0);
-            material.MetallicRoughnessMaterial = null;
+                // Create the gltf object
+                return new Model
+                {
+                    Properties = properties,
+                    GLTF = CreateGLTF(() => new Runtime.Scene()
+                    {
+                        Nodes = new List<Runtime.Node>
+                        {
+                            new Runtime.Node
+                            {
+                                Mesh = new Runtime.Mesh
+                                {
+                                    MeshPrimitives = new List<Runtime.MeshPrimitive>
+                                    {
+                                        meshPrimitive
+                                    }
+                                },
+                            },
+                        },
+                    }),
+                };
+            }
 
-            foreach (Property property in combo)
+            void SetVertexColorVec3Float(List<Property> properties, Runtime.MeshPrimitive meshPrimitive)
             {
-                wrapper.Scenes[0].Nodes[0].Mesh.MeshPrimitives[0].ColorComponentType = property.value.componentType;
-                wrapper.Scenes[0].Nodes[0].Mesh.MeshPrimitives[0].ColorType = property.value.type;
-                wrapper.Scenes[0].Nodes[0].Mesh.MeshPrimitives[0].Colors = property.value.colors;
+                meshPrimitive.ColorComponentType = Runtime.MeshPrimitive.ColorComponentTypeEnum.FLOAT;
+                meshPrimitive.ColorType = Runtime.MeshPrimitive.ColorTypeEnum.VEC3;
+
+                var vertexColorsValue = new VertexColor(meshPrimitive.ColorComponentType, meshPrimitive.ColorType, meshPrimitive.Colors);
+                properties.Add(new Property(PropertyName.VertexColor, vertexColorsValue));
+            }
+
+            void SetVertexColorVec3Byte(List<Property> properties, Runtime.MeshPrimitive meshPrimitive)
+            {
+                meshPrimitive.ColorComponentType = Runtime.MeshPrimitive.ColorComponentTypeEnum.NORMALIZED_UBYTE;
+                meshPrimitive.ColorType = Runtime.MeshPrimitive.ColorTypeEnum.VEC3;
+
+                var vertexColorsValue = new VertexColor(meshPrimitive.ColorComponentType, meshPrimitive.ColorType, meshPrimitive.Colors);
```

Comment:
I don't understand the point of the VertexColor class. Let's discuss offline.

Reply:
After discussion, just put a string here for the string we want to show for the readme.

Reply:
Fixed